File: src/main/java/com/revature/GameLogic/Blackjack/BlackjackClientGameState.java
package com.revature.GameLogic.Blackjack;

import java.util.List;

import com.revature.CardLogic.Card52;
import com.revature.GameLogic.AllGames.BaseClientGameState;
import com.revature.GameLogic.Blackjack.BlackjackPlayer.EndGameStates;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

public @Data @AllArgsConstructor @NoArgsConstructor class BlackjackClientGameState implements BaseClientGameState {
    private List<Card52> dealersCards;
    private List<BlackjackPlayerInfo> players;

    public static @Data @AllArgsConstructor @NoArgsConstructor class BlackjackPlayerInfo {
        EndGameStates endGameState;
        String playerName;
        boolean hasTakenTurn;
        List<Card52> cards;
    }
}

File: demo/src/model/BlackjackClientGameState.ts
import { BlackjackPlayerInfo } from "./BlackJackPlayerInfo";
import { Card52 } from "./Card52";

export class BlackjackClientGameState {
    dealersCards: Card52[];
    players: BlackjackPlayerInfo[];
    
    constructor(dealersCards: Card52[], players: BlackjackPlayerInfo[]) {
        this.dealersCards = dealersCards;
        this.players = players;
        
    }
}